src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,147 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> prevNode = tail;
+        Node<T> newNode = new Node<>(prevNode, value, null);
+        tail = newNode;
+        if (prevNode == null) {
+            head = newNode;
+        } else {
+            prevNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(
+                    "Index " + index + " is out of bounds " + size + "!");
+        }
+
+        if (index == size) {
+            add(value);
+        } else {
```

Comment:
You already have logic to check index. For example:
```suggestion
       if (index == size) {
           add(value);
       } else {
           checkIndex(index)
```